ext/ForwardDiffStaticArraysExt.jl

```diff
@@ -10,6 +10,10 @@ using ForwardDiff: Dual, partials, GradientConfig, JacobianConfig, HessianConfig
                    vector_mode_jacobian, vector_mode_jacobian!, valtype, value
 using DiffResults: DiffResult, ImmutableDiffResult, MutableDiffResult
 
+_chunk(::Length{l}, s::StaticArray) where {l} = Chunk{l}()
+
+ForwardDiff.Chunk(s::StaticArray) = _chunk(Length(s), s)
```

Comment:
How about we store the DEFAULT_CHUNK_THRESHOLD as a Val? It can then be used to compute the Chunk in a type-stable manner while respecting the threshold. It only changes DEFAULT_CHUNK_THRESHOLD, so it shouldn't be breaking (unless `ForwardDiff.DEFAULT_CHUNK_THRESHOLD` is a public API, which I don't think it is)

Reply:
Alright, my more recent attempt at least seems to respect the limit, though it does still look rather scary to me, see below.